### ISSUE TITLE
Side by side viewer

### ISSUE THREAD

**Issue by HarrisonMc555** [dandavison/delta#86]
Are there any plans to support side-by-side viewing?

**Comment by dandavison** [comment]
Can you expand on what this feature would be? I don't think it's been discussed yet. So far, delta is based around parsing default git output and unified diff output (which is very similar).

**Comment by HarrisonMc555** [comment]
Lots of IDEs and GUI diff tools show differences side-by-side, instead of on different lines in a unified output like `diff` (and `git diff`) do by output. In addition, there are several command-line tools, such as `sdiff` and [`icdiff`](https://github.com/jeffkaufman/icdiff).

Do you think you would ever consider something like that? It may be out of the scope of this project, which I would completely understand. But I figured I would ask 😄

**Comment by dandavison** [comment]
Hi @HarrisonMc555, thanks! OK, so, delta doesn't have access to the original files being compared: what delta is is a tool for applying syntax highlighting to the _output_ of diff programs (such as `git` and `diff -u`). In addition to simple syntax highlighting, it tries to do some fancy string-comparison edit inference, and restructures the git output a bit for readability.

So delta itself would not _generate_ the side-by-side view (it can't, it doesn't have access to the original files). Delta could however process the output of a program such as `sdiff` or `icdiff`, to add syntax highlighting etc. (Git can be configured to use an arbitrary executable for its diffs as explained in [this SO post](https://stackoverflow.com/a/7669988/583763)). This would require writing a new module for parsing those new sorts of inputs, so it would be quite a major addition, and I don't think I will be doing it personally.

OK, here's a question. I wonder whether, rather than using delta, there's a solution that uses `bat` to do the syntax highlighting in combination with `sdiff` or `icdiff`? These naive attempts didn't seem to work, but do you think something in this direction could work?

```
sdiff <(bat --plain --color=always a.py) <(bat --plain --color=always b.py)
icdiff <(bat --plain --color=always a.py) <(bat --plain --color=always b.py)
```

![image](https://user-images.githubusercontent.com/52205/72617857-e3f10800-3907-11ea-8483-35d661568992.png)

Ref https://github.com/jeffkaufman/icdiff/issues/133

**Comment by HarrisonMc555** [comment]
Thanks for your response! I'll have to think about it for a bit. Thanks for your help!

**Comment by dandavison** [comment]
No problem, I'll close this for now, but feel free to add any further thoughts.

**Comment by Dentrax** [comment]
@dandavison This feature would be so awesome with this amazing tool. `icdiff` have this side-by-side feature, we can also add a option like that called: `--side-by-side`, which would be so great. ;)

**Comment by dandavison** [comment]
Thanks @Dentrax and @HarrisonMc555. I think I said something wrong above. It's possible for delta to construct a side-by-side view without access to the original files. I agree that this is a feature that would very much make sense.

**Comment by panchalkalpesh** [comment]
+1 👍 
We (developers) often have a wider screen which can accommodate **side-by-side** view.
Definitely a nice feature to have as it increases the usability of this tool.

@Dentrax Thanks for sharing, quite interesting.

**Comment by mscottnelson** [comment]
@dandavison would you like any help with this issue? not making any promises but I'd be interested in looking into it

**Comment by dandavison** [comment]
@mscottnelson thanks, that would be great! So I'll give you a bit of time to take a look at what needs to be done -- I won't do anything in this area without pinging you on this ticket first.

I'm currently working on getting the next release out (0.2.0). But let's not try to include this feature in that release -- we can release this when it's ready.  I'm excited about this feature, but there's also no shortage of other delta improvements for me to work on, so very happy for others to take features like this. I believe that the recent changes in master have put us in a decent position to implement this. For example, line numbers have been added to delta by @clnoll in #190 (and open PR #219), and they will have a left and right column of numbers that should fit naturally into the left and right split in side-by-side view.

**Comment by dandavison** [comment]
@mscottnelson are you still thinking of working on this or should we consider it open again?

**Comment by dandavison** [comment]
@mscottnelson I'm (probably!) going to have a go at this. Let me know if you've already started work on it.

**Comment by mscottnelson** [comment]
@dandavison Great! sorry for the poor communication. Thanks for being so awesome/proactive!

**Comment by Dentrax** [comment]
Thank you so much for such a great job! @dandavison

**Comment by dandavison** [comment]
Side-by-side diff view has been released (delta 0.3.0). It has a [section in the README](https://github.com/dandavison/delta#side-by-side-view), but basically you use
```gitconfig
[delta]
    side-by-side = true
```
or the equivalent command-line flags `-s` or `--side-by-side`.

<table><tr><td><img width=800px src="https://user-images.githubusercontent.com/52205/87230973-412eb900-c381-11ea-8aec-cc200290bd1b.png" alt="image" /></td></tr>
<tr><td><img width=800px src="https://user-images.githubusercontent.com/52205/87232665-c5d40400-c38e-11ea-84e2-e84feff84149.png" alt="image" /></td></tr>
</table>


cc @Amorymeltzer, @Avi-D-coder, @Kr1ss-XD, @MarcoIeni, @RobbieClarken, @Valloric, @crash-g, @jeffkaufman, @jwatt, @kenchan0130, @kyohsuke, @lightspeedbriefs, @madnight, @maricn, @mosic, @mscottnelson, @navarroaxel, @nemonweb, @panchalkalpesh, @softoika, @wataash, @zdenekkostal

(Thanks @Dentrax!)

**Comment by navarroaxel** [comment]
This feature really rocks! :clap:
